brian2tools/nmlimport/nml.py

```diff
@@ -0,0 +1,349 @@
+import neuroml.loaders as loaders
+from neuroml.utils import validate_neuroml2
+from brian2 import Morphology
+from brian2.utils.logger import get_logger
+from brian2.spatialneuron.morphology import Section,Soma,Cylinder
+from brian2.units import *
+from os.path import abspath
+from .helper import *
+from copy import deepcopy
+import re
+logger = get_logger(__name__)
+
+
+class ValidationException(Exception):
+    pass
+
+
+def validate_morphology(segments):
+    """
+    Validates if the segments are connected to each other or not.
+
+    Parameters
+    ----------
+    segments : list
+        list of segments present in a morphology
+
+    Returns
+    -------
+    """
+
+    try:
+        start_segment = None
+        for segment in segments:
+
+            if segment.parent is not None:
+                if segment.parent.fraction_along not in [0,1]:
+                    raise NotImplementedError(
+                        "{0} has fraction along value {1} which is not "
+                        "supported!!".
+                        format(segment,segment.parent.fraction_along))
+
+                seg_parent = get_parent_segment(segment, segments)
+
+                if not are_segments_joined(segment, seg_parent):
+                    raise ValidationException(
+                        "{0} and its parent segment {1} are not connected!!".
+                        format(formatter(segment), formatter(seg_parent)))
+
+            elif start_segment is not None:
+                raise ValidationException(
+                    "Two segments with parent id as -1 (i.e no parent): "
+                    "{0} and {1}".
+                    format(formatter(start_segment), formatter(segment)))
+            else:
+                start_segment = segment
+                logger.info(
+                    "starting segment is {0}".format(formatter(segment)))
+
+    except ValidationException as v:
+        logger.error("Validation error: {0}".format(v))
+        raise
+
+    except Exception as e:
+        logger.error("Exception occured: {0}".format(e))
+        raise
+
+
+class NmlMorphology(object):
```

Comment:
For consistency with the NeuroML exporter, this should probably be called `NMLMorphology`